_rules/SC3-1-1-html-xml-lang-match.md

```diff
@@ -23,9 +23,9 @@ The root node of the page, if it is an `html` element with both `lang` and `xml:
 
 ### Expectation
 
-- The value of the primary language sub-tag for the `lang` and `xml:lang` attributes are the same.
+The value of the primary language sub-tag for the `lang` and `xml:lang` attributes are the same.
 
```

Comment:
I don't know why this is saying only the "primary language sub-tags" need to match. Why allow mismatches in the other sub-tags? There is no benefit to this and it is a violation of the HTML 5 specification. Just say "The values must be identical when compared case-insensitively".

Reply:
@jribbens This is testing compliance to WCAG 2.0, not to HTML 5. To the best of my knowledge, a mismatch between secondary languages does not lead to accessibility problems, unlike what can happen if the primary language sub-tag is incorrect.

Reply:
@WilcoFiers Clearly a secondary sub-tag mismatch is *less likely* to lead to accessibility problems than a primary mismatch, but stating outright that it *does not* seems brave. I can see no benefit to allowing mismatches, and it seems like erring on the side of incaution.

Reply:
Can you give an example of when getting the secondary sub-tag wrong can be a violation of WCAG? Even for example if you do "en-US" on a page with British spelling, that shouldn't be a violation of WCAG. Even something non-existent such as "en-XYZ" doesn't lead to a violation, so a mismatch between a secondary sub-tag shouldn't make a difference either.

Reply:
Well, `<span lang="en-us">This is a boot sale</span>` vs `<span lang="en-gb">This is a boot sale</span>` have quite different meanings, if that's the sort of thing you're looking for, or similarly with "This is the first floor" :-)

Honestly I don't know if there are languages where words might be mispronounced in a way as to make them ambiguous or incomprehensible if the wrong secondary sub-tag is used, it just seems to bizarre to make the test *more* complicated on the off-chance that *maybe* a situation that will almost never come up won't matter.

I do think there needs to be a discussion as to what the possible outcomes of a auto-wcag test can be... we have pass/fail/inapplicable but I think at least "possible fail" and "warning" are needed. Then if we do want this test to be complicated then it could have "fail" for primary mismatch and "warning" if not identical.

Reply:
Would you agree to covering this with an assumption, something like:

> Although there is no known way that an inappropriate secondary sub-tag in the `lang` or `xml:lang` attribute (such as "en-XYZ") can lead to violations of 3.1.2, it is conceivable that with specific assistive technologies on some languages, there can be exceptions. In that case `lang` and `xml:lang` should have matching secondary sub-tags.

Reply:
It's not. Assumptions are just that, things that weren't addressed by the rule that may lead to incorrect results. It's almost impossible to write a rule that's entirely air tight, so where it isn't, we include that information.